src/main/java/seedu/address/logic/parser/DeleteCommandParser.java

```diff
@@ -18,12 +19,16 @@ public class DeleteCommandParser implements Parser<DeleteCommand> {
      */
     public DeleteCommand parse(String args) throws ParseException {
         try {
-            Index index = ParserUtil.parseIndex(args);
-            return new DeleteCommand(index);
+            if (args.contains("n/")) {
+                Name name = ParserUtil.parseName(args.replace("n/", "").trim());
+                return new DeleteCommand(name);
+            } else {
+                Index index = ParserUtil.parseIndex(args);
+                return new DeleteCommand(index);
+            }
         } catch (ParseException pe) {
             throw new ParseException(
-                    String.format(MESSAGE_INVALID_COMMAND_FORMAT, DeleteCommand.MESSAGE_USAGE), pe);
+                    String.format(MESSAGE_INVALID_COMMAND_FORMAT, DeleteCommand.MESSAGE_USAGE));
```

Comment:
```suggestion
                    String.format(MESSAGE_INVALID_COMMAND_FORMAT, DeleteCommand.MESSAGE_USAGE), pe);
```

Reply:
Perhaps you can consider including the ParseException pe as there might be extra information contained within that needs to output in the exception message.